lib/net.js

```diff
@@ -256,6 +258,9 @@ function Socket(options) {
 
   // Used after `.destroy()`
   this[BYTES_READ] = 0;
+
+  this[kLastTimeout] = 0;
+  this[kPrevTimeout] = 0;
```

Comment:
I'm not extremely happy with adding these properties to `Socket` too, but I don't have any other solution offhand. The problem is that the HTTP server needs the previous value of the timeout to restore it after the keep-alive timeout is not needed anymore, but it does not necessarily know what the value is. `test-https-set-timeout-server` used to fail before f381340eaff45e840f6a35a954333dc4b41bca57 because of using `res.setTimeout()`, which, well, may be resolved at the HTTP level, but a user could have used `res.socket.setTimeout()` directly, and there's not much we can in `http` in this case.